### PR TITLE
feat(storage-browser): add/update/refactor view and controls components

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
- add missing `View` and `Control` components
- update composable hierarchy
- add `MergeBaseElement` util, addresses uninstantiated type errors
- add placeholder unit tests and missing test files

#### Issue #, if available
NA

#### Description of how you validated changes
Manually tested, init unit tests
#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] `yarn test` passes and tests are updated/added
- [x] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by reesscot — APPROVED** [review]

File: packages/react-storage/src/components/StorageBrowser/Views/LocationDetail/LocationDetailView.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+export interface LocationDetailView<
+  _T extends StorageBrowserElements = StorageBrowserElements,
+> {
+  (): React.JSX.Element;
+  ActionView: ActionView;
+  ListView: ListView;
+}
+
+export const LocationDetailView: LocationDetailView = () => {
+  const [{ action }] = useControlState?.('ACTION_SELECT') ?? [
```
[the remark this answers is not in the document]
Comment (calebpollman) [reply packages/react-storage/src/components/StorageBrowser/Views/LocationDetail/LocationDetailView.tsx:18]: Intentional, currently used for manual testing

File: packages/react-storage/src/components/StorageBrowser/createStorageBrowser.tsx [hunk 2]
```diff
@@ -1,56 +1,52 @@
...
-  TitleControl,
-} from './Views';
-import { Controls, CreateStorageBrowserInput, StorageBrowser } from './types';
-
-export default function createStorageBrowser<T extends StorageBrowserElements>({
-  elements: _elements,
-}: CreateStorageBrowserInput<Partial<T>> = {}): {
-  StorageBrowser: StorageBrowser<T>;
+import { LocationsListView, LocationDetailView } from './Views';
+import { CreateStorageBrowserInput, StorageBrowser } from './types';
```
[the remark this answers is not in the document]
Comment (calebpollman) [reply packages/react-storage/src/components/StorageBrowser/createStorageBrowser.tsx]: Update offending test code to remove error. Test code is currently used for catching type errors

File: packages/react-storage/src/components/StorageBrowser/Views/LocationsListView/LocationsListView.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+const LocationsListViewControls: LocationsListViewControls = () => (
+  <div className={`${CLASS_BASE}__header`}>
+    <Navigate />
+    <div className={`${CLASS_BASE}__header__primary`}>
+      <Title />
+      <div className={`${CLASS_BASE}__header__primary__actions`}>
+        <Refresh />
+      </div>
+    </div>
+    <div className={`${CLASS_BASE}-header__secondary`}>
```
Comment (timngyn) [inline packages/react-storage/src/components/StorageBrowser/Views/LocationsListView/LocationsListView.tsx:35]: nit: Should this use the `__` after CLASS_BASE, like the other classes you have above?

**Review by timngyn — APPROVED** [review]

File: packages/react-storage/src/components/StorageBrowser/Views/LocationsListView/LocationsListView.tsx (hunk 3 above)
Comment (calebpollman) [reply packages/react-storage/src/components/StorageBrowser/Views/LocationsListView/LocationsListView.tsx:35]: @hbuchel thoughts? @timngyn  this code was migrated and updated minimally

**Pull request merged**